nighthawk/source/client/benchmark_client_impl.cc

```diff
@@ -0,0 +1,243 @@
+#include "nighthawk/source/client/benchmark_client_impl.h"
+
+#include "absl/strings/str_split.h"
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/thread_local/thread_local.h"
+
+#include "extensions/transport_sockets/well_known_names.h"
+
+#include "common/common/compiler_requirements.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/http/http1/conn_pool.h"
+#include "common/http/http2/conn_pool.h"
+#include "common/http/utility.h"
+#include "common/network/dns_impl.h"
+#include "common/network/raw_buffer_socket.h"
+#include "common/network/utility.h"
+#include "common/upstream/cluster_manager_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+#include "nighthawk/common/statistic.h"
+
+#include "nighthawk/source/client/stream_decoder.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+namespace Client {
+
+BenchmarkClientHttpImpl::BenchmarkClientHttpImpl(Envoy::Api::Api& api,
+                                                 Envoy::Event::Dispatcher& dispatcher,
+                                                 Envoy::Stats::StorePtr&& store,
+                                                 StatisticPtr&& connect_statistic,
+                                                 StatisticPtr&& response_statistic,
+                                                 const std::string& uri, bool use_h2)
```

Comment:
Tiny nit: generally prefer `absl::string_view` over `const std::string&` in new code, since it's strictly more flexibly and compatible with the C++17 brace new world.

Reply:
Yeah, added a tracking issue to replace the remaining `const std::string&` instances with `absl::string_view` in one go; resolving this for now.